csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/SourceGenerators/DotnetSourceGeneratorWrapper/DotnetSourceGeneratorWrapper.cs

```diff
@@ -35,23 +35,22 @@ internal abstract class DotnetSourceGeneratorWrapper
 
         public IEnumerable<string> RunSourceGenerator(IEnumerable<string> additionalFiles, string csprojFile, IEnumerable<string> references, string targetDir)
         {
-            var name = Guid.NewGuid().ToString("N").ToUpper();
-            var tempPath = FileUtils.GetTemporaryWorkingDirectory(out var shouldCleanUp);
-            var analyzerConfig = Path.Combine(tempPath, $"{name}.txt");
-            var dllPath = Path.Combine(tempPath, $"{name}.dll");
-            var cscArgsPath = Path.Combine(tempPath, $"{name}.rsp");
-            var outputFolder = Path.Combine(targetDir, name);
-            Directory.CreateDirectory(outputFolder);
-
             try
             {
+                var name = Guid.NewGuid().ToString("N").ToUpper();
+                using var tempDir = new TemporaryDirectory(Path.Join(FileUtils.GetTemporaryWorkingDirectory(out _), "source-generator"), "source generator temporary", logger);
```

Comment:
`tempDir` is going to be inside the DB scratch folder or the system temp dir. Regardless of the `shouldCleanUp`, it will always be cleaned up, because of the `using var` declaration. I think it's okay to clean this subfolder up, even if `shouldCleanUp` says otherwise. We're already doing this for some other `TemporaryDirectory`s.